contents/testing/integration-testing.md

```diff
@@ -0,0 +1,155 @@
+<frontmatter>
+  title: Integration Testing
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Integration Testing
+
+**Author(s): [Jacob Li PengCheng](https://github.com/jacoblipech)**
+
+Reviewers:
+
+## What is integration testing?
+
+Integration testing is a part of software testing where individual parts of your application are combined and tested as a whole. This usually happens after the [<tooltip content="Testing of individual components within a system">unit testing stage</tooltip>](http://softwaretestingfundamentals.com/unit-testing/) and before the [<tooltip content="Evaluation of the software against requirements gathered from users and system specifications">system testing stage</tooltip>](https://www.guru99.com/system-testing.html) and it aims to discover faults related to the interaction between components.
+
+## Why is integration testing important?
+
+In a software application, each software module is usually designed and unit tested by different programmers. Since these programmers may work in isolation and have different understandings of the software requirements, integration testing is necessary to ensure that software modules work in unity and expose any faults in the interaction between different units.
+
+<box>Suppose you are in charge of a **data collection component** that passes the **data** collected from users to a **data analysis component**. If the **data analysis component** assumes your **data** to be of maximum 100 lines but the data you send has 200 lines, it may result in data loss, inaccurate results and even system errors. Integration tests find these discrepancies before any serious damage is done.</box>
+
+Hence, integration testing is an important process which should be applied in the development of most softwares products such as web, mobile and desktop applications. Concepts shared in this article can be adopted by other different software applications.
```

Comment:
The reader might wonder why we need integration testing when the same kind of problems can also be discovered via system testing.